protocol/xmrtaker/message_handler.go

```diff
@@ -148,6 +143,52 @@ func (s *swapState) handleSendKeysMessage(msg *message.SendKeysMessage) (common.
 	return out, nil
 }
 
+func (s *swapState) checkForXMRLock() {
+	// monero block time is >1 minute, so this should be fine
+	const checkForXMRLockInterval = time.Minute
+
+	// check that XMR was locked in expected account, and confirm amount
+	lockedAddr, vk := s.expectedXMRLockAccount()
+
+	conf := s.XMRClient().CreateWalletConf("xmrtaker-swap-wallet-verify-funds")
+	abViewCli, err := monero.CreateViewOnlyWalletFromKeys(conf, vk, lockedAddr, s.walletScanHeight)
+	if err != nil {
+		log.Errorf("failed to generate view-only wallet to verify locked XMR: %s", err)
+		return
+	}
+	defer abViewCli.CloseAndRemoveWallet()
+
+	log.Debugf("generated view-only wallet to check funds: %s", abViewCli.WalletName())
+
+	timer := time.NewTimer(checkForXMRLockInterval)
```

Comment:
We have a `SleepWithContext` method that might be simpler.

Reply:
I updated this to use a ticker, left it as is since I think it's a bit more readable and explicit